plugins/manipulator.cpp

```diff
@@ -304,144 +352,393 @@ struct UnitInfo
-        df::unit_skill *s1 = binsearch_in_vector<df::unit_skill,df::job_skill>(d1->unit->status.current_soul->skills, &df::unit_skill::id, sort_skill);
-        df::unit_skill *s2 = binsearch_in_vector<df::unit_skill,df::job_skill>(d2->unit->status.current_soul->skills, &df::unit_skill::id, sort_skill);
-        int l1 = s1 ? s1->rating : 0;
-        int l2 = s2 ? s2->rating : 0;
-        int e1 = s1 ? s1->experience : 0;
-        int e2 = s2 ? s2->experience : 0;
-        if (descending)
-        {
-            if (l1 != l2)
-                return l1 > l2;
-            if (e1 != e2)
-                return e1 > e2;
-        }
-        else
-        {
-            if (l1 != l2)
-                return l1 < l2;
-            if (e1 != e2)
-                return e1 < e2;
-        }
+        l1 = unitSkillRating(d1,sort_skill)*50;
+        l1 += 25+unitSkillExperience(d1,sort_skill)*40/(skill_levels[l1/50].points+1);
+        //eg skill 1 = 50, 5 = 250, 10 = 500
+        //l1 += d1->column_aptitudes[column_sort_column];
+        l1 *= d1->column_aptitudes[column_sort_column]+300;
+
+        l2 = unitSkillRating(d2,sort_skill)*50;
+        l2 += 25+unitSkillExperience(d2,sort_skill)*45/(skill_levels[l2/50].points+1);
+        //l2 += d2->column_aptitudes[column_sort_column];
+        l2 *= d2->column_aptitudes[column_sort_column]+300;
     }
-    if (sort_labor != unit_labor::NONE)
-    {
-        if (descending)
-            return d1->unit->status.labors[sort_labor] > d2->unit->status.labors[sort_labor];
-        else
-            return d1->unit->status.labors[sort_labor] < d2->unit->status.labors[sort_labor];
+
+    if(sort_labor != unit_labor::NONE){
+        if(d1->unit->status.labors[sort_labor])
+            l1+= 10+l1/10;
+        if(d2->unit->status.labors[sort_labor])
+            l2+= 10+l2/10;
     }
-    return false;
+
+    return l1 > l2;
+
 }
 
-bool sortBySelected (const UnitInfo *d1, const UnitInfo *d2)
+bool sortByUnitId (const UnitInfo *d1, const UnitInfo *d2 ){
+    return (d1->unit->id > d2->unit->id);
+}//used to find arrival_groups
+
+bool sortByActiveIndex (const UnitInfo *d1, const UnitInfo *d2 ){
+    return (d1->active_index > d2->active_index);
+}//used to find arrival_groups
+
+
+string itos (int n)
+{
+    stringstream ss;
+    ss << n;
+    return ss.str();
+}
+
+PersistentDataItem config_dfkeeper;
+void save_dfkeep_config()
+{
+    config_dfkeeper = World::GetPersistentData("dfkeeper/config");
+    if (!config_dfkeeper.isValid()){
+        config_dfkeeper = World::AddPersistentData("dfkeeper/config");
+        if (!config_dfkeeper.isValid())
+            return;
+    }
+
+    config_dfkeeper.ival(0) = color_mode+1;
+    config_dfkeeper.ival(1) = hint_power;
+    config_dfkeeper.ival(2) = spare_skill;
+    config_dfkeeper.ival(3) = show_details+1;
+    config_dfkeeper.ival(4) = tran_names;
+    config_dfkeeper.ival(5) = 0;
+    config_dfkeeper.ival(6) = 0;
+}
+
+void read_dfkeep_config()
```

Comment:
This doesn't really matter, but maybe rename read_dfkeep_config() to read_dfkeeper_config(), save_dfkeep_config() to save_dfkeeper_conifg() and dfkeep_palette to dfkeeper_palette?